### PR TITLE
fix: Fully qualifies all alternative types/members for obsolete ones.

### LINKED ISSUE #6672: CloudTraceExtensions.TryAddGoogleTraceContextProvider is missing
Hi,

I upgraded to Google.Cloud.Diagnostics.AspNetCore3 4.3.0-beta01 and have had to make a few changes to my code to accommodate various obsoletions and changes.

I have this code in my `ConfigureServices` method:
`services.AddSingleton<ITraceContext>(new GoogleTraceProvider());`

In the beta this no longer compiles. It tells me to: "Use CloudTraceExtensions.TryAddGoogleTraceContextProvider instead."

That method does not seem to exist.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Fixes #6672

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
Failures are because of the diffs that we accepted on #6657.

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/IExternalTraceProvider.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ namespace Google.Cloud.Diagnostics.AspNetCore
     /// Implement this interface when traces are being generated by your environment
     /// and sent to Google Cloud Tracing not via the Google.Cloud.Diagnostics library.
     /// </summary>
-    [Obsolete("Inject an Common.ITraceContextInstead.")]
+    [Obsolete("Inject an Google.Cloud.Diagnostics.Common.ITraceContext instead.")]
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/IExternalTraceProvider.cs:31]: Nit: an => a

**Review by jskeet — APPROVED** [review]
One nit

**Comment by amanda-tarafa** [comment]
Nit fixed! You'll have to merge because of the diff.

**Comment by jskeet** [comment]
Righto - merging now.

**Pull request merged**